ouroboros-network/src/Ouroboros/Network/BlockFetch.hs

```diff
@@ -115,14 +114,21 @@ data BlockFetchConsensusInterface peer header block m =
        -- They must be already validated and contain the last @K@ headers
        -- (unless we're near the chain genesis of course).
        --
-       readCandidateChains    :: STM m (Map peer (ChainFragment header)),
+       -- Each candidate chain will be anchored within the bounds
+       -- ('AnchoredFragment.withinFragmentBounds') of the current chain
+       -- ('readCurrentChain'), which means that there will be an intersection
+       -- between the candidate chain and the current chain. In other words,
+       -- each candidate forks off within the last @K@ blocks of the current
+       -- chain (or less when we're near genesis).
```

Comment:
We should clarify that _typically_ the candidate chains will be anchored within the current chain. But it's ok for them to (temporarily) not be, and the fetch logic will ignore them in that situation since there's no intersection.

Ie we should clarify that it's not a precondition that at all times the anchor point be within the current chain, it's just a typical case and needed to make progress with fetching anything.